docs/rules.md

```diff
@@ -13,215 +27,214 @@ A Rule is written in YAML. It consists of metadata, conditions and actions. It i
+Analyzer currently supports `builtin`, `java`, `go` and `generic` providers. Here is the table that summarizes all the providers and their capabilities:
 
-###### referenced
+| Provider Name | Capabilities                                                  | Description                                                                       |
+| ------------- | ------------------------------------------------------------- | --------------------------------------------------------------------------------- |
+| java          | referenced                                                    | Find references of a pattern with an optional code location for detailed searches |
+|               | dependency                                                    | Check whether app has a given dependency                                          |
+| builtin       | xml                                                           | Search XML files using xpath queries                                              |
+|               | json                                                          | Search JSON files using jsonpath queries                                          |
+|               | filecontent                                                   | Search content in regular files using regex patterns                              |
+|               | file                                                          | Find files with names matching a given pattern                                    |
+|               | hasTags                                                       | Check whether a tag is created for the app via a tagging rule                     |
+| go            | referenced                                                    | Find references of a pattern                                                      |
+|               | dependency                                                    | Check whether app has a given dependency                                          |
 
-`referenced` capability enables the provider to find references in the source code. It takes two input parameters:
+Based on the table above, we should be able to create the first part of the condition that doesn’t contain any of the condition fields. For instance, to create a `java` provider condition that uses `referenced` capability:
 
 ```yaml
 when:
   java.referenced:
-    # regex pattern to match
-    pattern: "<pattern>"
-    # location defines the exact location where
-    # pattern should be matched
-    location: CONTRUCTOR_CALL
+    <fields>
 ```
 
-The supported locations are:
-- CONSTRUCTOR_CALL
-- TYPE
-- INHERITANCE
-- METHOD_CALL
-- ANNOTATION
-- IMPLEMENTS_TYPE
-- ENUM_CONSTANT
-- RETURN_TYPE
-- IMPORT
-- VARIABLE_DECLARATION
+Depending on the _provider_ and the _capability_, there will be different `<fields>` in the condition. Following table summarizes available providers, their capabilities and all of their fields:
```

Comment:
We need this to by dynamic in the future, and I would love a part of the UI to be able to display this, based on providers that are available to the user.